### DISCUSSION TITLE
Generating an In-Silico Library

### DISCUSSION THREAD

**Discussion by lifeminetx** [vdemichev/DiaNN#608] [Q&A] [answered]
Hi,

We are new to using DIA-NN and we started by generating an in-silico library using a FASTA file that was downloaded off SwissProt.
The FASTA contains 20389 reviewed proteins but DIA-NN only registers 5029 for in-silico digestion. 

Does anyone have an advice on how to get DIA-NN to include all proteins in the library. 

Many thanks
[DIA-NN IN-SILICO Digest Example.txt](https://github.com/vdemichev/DiaNN/files/10574069/DIA-NN.IN-SILICO.Digest.Example.txt)

**Comment by vdemichev** [comment, marked as answer]
In the log attached DIA-NN says it detects 20370 proteins. This refers not to sequence IDs (the number of which might be higher) but to sequence IDs which have UniProt names provided (which are in the formant e.g. [PROTEIN]_HUMAN).

**Reply by lifeminetx** [reply]
Many thanks Vadim!
